### PR TITLE
Allow passing URLSession to StarknetProvider

### LINKED ISSUE #120: Extend `StarknetProvider` to allow passing an initialized `URLSession` instance
Clients should have better control over the network stack to manage components such as timeouts, additional headers, and other settings relevant to the underlying product. `StarknetProvider` should not implicitly create a default `URLSession`; instead, it should have a new initializer to receive a pre-configured instance of `URLSession` and use that one. If not provided, it should initialize a default one, as is currently done.

### PR DESCRIPTION
**Pull request by DelevoXDG**
### Changes
- #120 
- Format
- Bumped devnet
  - Fix broken DevnetClient test
  - Disabled som tests that got broken (tracking in #89) 
- Install sncast using asdf
  
### Closed issues
Closes #120

### CODE REVIEW

File: Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift [hunk 1]
```diff
@@ -25,6 +25,19 @@ public class StarknetProvider: StarknetProviderProtocol {
         self.init(starknetChainId: starknetChainId, url: url)
     }
 
+    public init(starknetChainId: StarknetChainId, url: URL, urlSession: URLSession) {
```
Comment (DelevoXDG) [inline Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift:28]: I also considered the solution with default parameter, but I think it's better if the default case (no URL session) is handled by `HttpNetworkProvider` itself

File: Tests/StarknetTests/Providers/ProviderTests.swift [hunk 2]
```diff
@@ -31,6 +31,15 @@ final class ProviderTests: XCTestCase {
...
     }
 
+    func testRequestWithCustomURLSession() {
+        let starknetChainId = StarknetChainId.testnet
+        let url = Self.devnetClient.rpcUrl
+        let customURLSession = URLSession(configuration: .ephemeral)
+        let starknetProvider = StarknetProvider(starknetChainId: starknetChainId, url: url, urlSession: customURLSession)
+
+        XCTAssertNotNil(starknetProvider)
+    }
```
Comment (DelevoXDG) [inline Tests/StarknetTests/Providers/ProviderTests.swift:41]: Added example test, but there's no need for it. This is just another constructor, so not much to test here.

**Review by THenry14 — APPROVED** [review]

File: Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift [hunk 3]
```diff
@@ -25,6 +25,19 @@ public class StarknetProvider: StarknetProviderProtocol {
...
+        self.starknetChainId = starknetChainId
+        self.url = url
+        self.networkProvider = HttpNetworkProvider(session: urlSession)
+    }
+
+    public convenience init?(starknetChainId: StarknetChainId, url: String, urlSession: URLSession) {
```
Comment (dmcrodrigues) [inline Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift:34]: Is this really needed? Provides convenience to consumers but then they will need to an optional provider anyway.

Comment (DelevoXDG) [reply Sources/Starknet/Providers/StarknetProvider/StarknetProvider.swift:34]: It's should it be there, at least for consistency

**Pull request merged**